### PR TITLE
design: add provider strategy doc

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

This doc discusses what to do with new Terrajet-based providers and native providers. All feedback, use cases, user stories are highly appreciated!

For TLDR, you can skip to **`Options`** section.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

N/A

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by negz — APPROVED** [review]
Thanks @muvaf! We discussed this synchronously and I feel good about where it's at.

File: design/design-doc-provider-strategy.md [hunk 1]
```diff
@@ -0,0 +1,746 @@
...
+follows when it comes what tooling the provider is using. For example, Google
+provider uses DCL and SDK at the same time.
+
+## Options
+
+### A: Provider per SDK/API
+
+We can have the following set of providers to cover three big clouds:
+* AWS
+  * Terrajet AWS: https://github.com/crossplane-contrib/provider-tf-aws
```
Comment (luebken) [inline design/design-doc-provider-strategy.md:506]: Should be https://github.com/crossplane-contrib/provider-jet-aws

File: design/design-doc-provider-strategy.md [hunk 2]
```diff
@@ -0,0 +1,746 @@
...
+implementation changes. For example, there can be a bug in Azure TF provider
+that we can't solve and we might want to switch to native implementation. The
+choice we make here should lower the cost of this switch as much as possible
+with the best user experience.
+
+#### B: Separation on Group Level
+
+We can separate the CRDs by the tool their managed reconciler uses on group
+level like:
+* `s3.aws.tf.crossplane.io/v1alpha1` for Terrajet S3
```
Comment (luebken) [inline design/design-doc-provider-strategy.md:560]: Should be * `s3.aws.jet.crossplane.io/v1alpha1` for Terrajet S3

**Comment by negz** [comment]
@muvaf I'll merge this once @luebken's updates have been addressed.

**Pull request merged**